### ISSUE TITLE
NPE (Ant-Deckard)

### ISSUE THREAD

**Issue by dmazinanian** [tsantalis/JDeodorant#33]
Calling PDGExpressionGap#getReturnType() for the only expression gap in the following clone pair returns Null. Consequently, CloneInfoCSVWriter is throwing NPE.

Clone # 1 (Group 352, Pair 1-2): org.apache.tools.ant.filters.HeadFilter#initialize (row 673) and org.apache.tools.ant.filters.TailFilter#initialize (row 674)

```
java.lang.NullPointerException
    at ca.concordia.jdeodorant.eclipse.commandline.cloneinfowriter.CloneInfoCSVWriter.writeCloneInfo(CloneInfoCSVWriter.java:156)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.testRefactoring(Application.java:757)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.start(Application.java:184)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1492)
```

**Comment by tsantalis** [comment]
Compilation error fixed in commit [0631698a484b23ac70afb26d0aae28aa85ba6d03](https://github.com/tsantalis/JDeodorant/commit/0631698a484b23ac70afb26d0aae28aa85ba6d03)

**Comment by tsantalis** [comment]
This case needs more careful examination, because there is no Lambda expression actually introduced.
The difference is a field assignment replaced with setter method invocation, and thus there is no parameterization needed, because the differences are semantically equivalent.
However, one instance of PDGExpressionGap remains in the list of refactorable expression gaps.

**Comment by tsantalis** [comment]
ExpressionGap.getReturnType() fixed in commit [d9ccf8391839f16d866c824bbb9f41efc39f455f](https://github.com/tsantalis/JDeodorant/commit/d9ccf8391839f16d866c824bbb9f41efc39f455f)
